spec/services/create_newsletter_signup_spec.rb

```diff
@@ -0,0 +1,37 @@
+require "rails_helper"
+require "test_helper"
+
+RSpec.describe CreateNewsletterSignup do
+
+  describe "newsletter signup object" do
+    it "accepts a first name, last name, and email" do
+      signup = CreateNewsletterSignup.new(
+        {
+          first_name: "Will",
+          last_name: "Blake",
+          email: "test@test.com"
+        }
+      )
+
+      expect(signup.newsletter_signup[:first_name]).to eq('Will')
+      expect(signup.newsletter_signup[:last_name]).to eq('Blake')
+      expect(signup.newsletter_signup[:email]).to eq('test@test.com')
+    end
+  end
+
+  describe "#perform" do
+    it "sends a signup request to SendGrid api" do
+      stub_request(:put, "https://api.sendgrid.com/v3/marketing/contacts").
+        with(
+          body: "{\"list_ids\":[\"c3f8fd1f-aea4-4611-8e7b-89bf8e12829a\"],\"contacts\":[{\"first_name\":\"Will\",\"last_name\":\"Blake\",\"email\":\"test@test.com\"}]}",
+          headers: {
+        'Accept'=>'application/json',
+        'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3',
+        'Authorization'=>'Bearer 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',
```

Comment:
This looks like a real token, still. Gonna need to remove & replace. We may need to pair here. All you gotta do is stub the key used. You can use this with ENV vars, or, with [dependancy injection](https://www.freecodecamp.org/news/a-quick-intro-to-dependency-injection-what-it-is-and-when-to-use-it-7578c84fa88f/)

Reply:
More specific detail: set the token as the default value of a KW arg for the service.